test/compiler/inline.jl

```diff
@@ -1375,3 +1375,23 @@ let src = code_typed1() do
     @test count(isnew, src.code) == 1
     @test count(isinvoke(:noinline_finalizer), src.code) == 1
 end
+
+# optimize `push!(::Vector{Any}, x...)`
+let src = code_typed1(push!, (Vector{Any}, Any))
+    @test count(iscall((src, push!)), src.code) == 0
+    @test count(src.code) do @nospecialize x
+        isa(x, Core.GotoNode) || isa(x, Core.GotoIfNot)
+    end == 0 # the loop should be optimized away for a single item
```

Comment:
this seems to be failing on many CI bots

Reply:
It seems like `push!(::Vector{Any}, ::Any)` produces the following IR:
```julia
julia> code_typed((Vector{Any},Any)) do xs, x
           push!(xs, x)
       end
1-element Vector{Any}:
 CodeInfo(
1 ─ %1 = Base.arraylen(xs)::Int64
│        $(Expr(:foreigncall, :(:jl_array_grow_end), Nothing, svec(Any, UInt64), 0, :(:ccall), Core.Argument(2), 0x0000000000000001, 0x0000000000000001))::Nothing
└──      goto #3 if not true
2 ─ %4 = Base.add_int(%1, 1)::Int64
└──      Base.arrayset(true, xs, x, %4)::Vector{Any}
3 ┄      goto #4
4 ─      return xs
) => Vector{Any}
```

So the loop can't be fully optimized on Julia level actually, but I confirmed LLVM can further optimize it down the road and the performance doesn't regress.